deps/sqlite/sqlite.gyp

```diff
@@ -3,26 +3,51 @@
     'sqlite_sources': [
       'sqlite3.c',
     ],
+    'use_system_sqlite%': 0,
   },
-  'targets': [
-    {
-      'target_name': 'sqlite',
-      'type': 'static_library',
-      'cflags': ['-fvisibility=hidden'],
-      'xcode_settings': {
-        'GCC_SYMBOLS_PRIVATE_EXTERN': 'YES',  # -fvisibility=hidden
-      },
-      'defines': [
-        'SQLITE_ENABLE_SESSION',
-        'SQLITE_ENABLE_PREUPDATE_HOOK'
+  'conditions': [
+    ['use_system_sqlite==0', {
```

Comment:
If you introduce this, something would need to toggle the value.

If configured with `--shared-sqlite`, this gyp file isn't included anyway (I verified that as part of https://github.com/nodejs/node/pull/55409 (you can remove `deps/sqlite` if using `--shared-sqlite` and the build still works)) as per https://github.com/nodejs/node/blob/6190bbc12c13705bc8003791e20f5b3742e27950/node.gypi#L231-L233 -- if needed perhaps any addition defines should be set there?

Reply:
Hum yeah that makes sense. For reference, I'm opening this because I'm getting `undefined reference to `sqlite3session_patchset'` when working on https://github.com/NixOS/nixpkgs/pull/357699, and Nix is using `--shared-sqlite`